core/src/subgraph/instance_manager.rs

```diff
@@ -739,6 +755,15 @@ where
                 proof_of_indexing.cheap_clone(),
             )
             .await
+            .map_err(|e| {
+                // This treats a `PossibleReorg` as an ordinary error which will fail the subgraph.
+                // This can cause an unecessary subgraph failure, to fix it we need to figure out a
+                // way to revert the effect of `create_dynamic_data_sources` so we may return a
+                // clean context as in b21fa73b-6453-4340-99fb-1a78ec62efb1.
```

Comment:
This feels like a tradeoff worth making to get the feature in. It also seems like something we should prioritize immediately after merging. I'm pretty sure the problem will happen, and it'll surface in a similar way to the current issues caused by calling by block.

Reply:
Considering how these triggers are much less common, the chance of us hitting the bug here is proportionally lower. And this bug already doesn't come up everyday, so I wouldn't count on us seeing this happen. But I think the fix for this and for the duplicate trigger processing will be similar, so we can try to fix them together.